packages/bolt-connection/src/connection-provider/connection-provider-routing.js

```diff
@@ -473,19 +478,19 @@ export default class RoutingConnectionProvider extends PooledConnectionProvider
         }
 
         // try next router
-        const session = await this._createSessionForRediscovery(
+        const [session, error] = await this._createSessionForRediscovery(
           currentRouter,
           bookmarks,
           impersonatedUser
         )
         if (session) {
```

Comment:
Generally (not only here), the readability might benefit from short-circuiting along the lines of

```typescript
const [foo, error] = await foobar();
if (error) { // alternatively: `if (foo === null) {` or similar
  return [null, error]
}
// ...
```

Reply:
Yes, but I'd like to change that much the code for this fix.